src/service/chat.js

```diff
@@ -206,6 +206,58 @@ const moveChatToAnon = async ({
   }
 };
 
+const initChatFromPost = async ({source, id}) => {
+  let payload = {source};
+  switch (source) {
+    case 'post':
+      payload = {...payload, postId: id};
+      break;
+    case 'comment':
+      payload = {...payload, commentId: id};
+      break;
+    default:
+      break;
+  }
+
+  try {
+    const response = await api.post('/chat/init-chat-from-post', payload);
+    if (response?.status === 200) {
+      return Promise.resolve(response.data);
+    }
+
+    return Promise.reject(response.data);
+  } catch (e) {
+    if (e?.response?.data?.message) return Promise.reject(e?.response?.data?.message);
+    return Promise.reject(e);
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `initChatFromPost` function seems to be well implemented. It correctly handles the payload creation based on the source, makes an API call, and handles the response as well as errors. However, it would be better to handle the default case in the switch statement to throw an error if an unexpected source is provided.

```diff
  switch (source) {
    case 'post':
      payload = {...payload, postId: id};
      break;
    case 'comment':
      payload = {...payload, commentId: id};
      break;
+   default:
+     throw new Error(`Unexpected source: ${source}`);
  }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->